### PR TITLE
.NET 8 compat test adjustments: 1) do not trim SDK, 2) support pattern to match output, 3) modify output truncation length

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: src/Runner.Worker/OSWarningChecker.cs [hunk 1]
```diff
@@ -83,14 +96,15 @@ public async Task CheckOSAsync(IExecutionContext context)
...
                 Trace.Error(ex);
-                context.Global.JobTelemetry.Add(new JobTelemetry() { Type = JobTelemetryType.General, Message = $".NET 8 OS compatibility test encountered exception type '{ex.GetType().FullName}', message: '{ex.Message}', process output: '{GetShortOutput(output)}'" });
+                context.Global.JobTelemetry.Add(new JobTelemetry() { Type = JobTelemetryType.General, Message = $".NET 8 OS compatibility test encountered exception type '{ex.GetType().FullName}', message: '{ex.Message}', process output: '{GetShortOutput(context, output)}'" });
             }
         }
 
-        private static string GetShortOutput(List<string> output)
+        private static string GetShortOutput(IExecutionContext context, List<string> output)
         {
+            var length = context.Global.Variables.System_DotNet8CompatibilityOutputLength ?? 200;
```
Comment (TingluoHuang) [inline src/Runner.Worker/OSWarningChecker.cs:105]: given we knows that 200 is not enough, should we bump it higher by default? :D

File: src/TestDotNet8Compatibility/TestDotNet8Compatibility.csproj [hunk 2]
```diff
@@ -6,7 +6,6 @@
     <RuntimeIdentifiers>win-x64;win-x86;linux-x64;linux-arm64;linux-arm;osx-x64;osx-arm64;win-arm64</RuntimeIdentifiers>
     <SelfContained>true</SelfContained>
     <PublishTrimmed>true</PublishTrimmed>
```
Comment (TingluoHuang) [inline src/TestDotNet8Compatibility/TestDotNet8Compatibility.csproj:8]: removed the wrong one?

Comment (ericsciple) [reply src/TestDotNet8Compatibility/TestDotNet8Compatibility.csproj:8]: oh snap, good catch

File: src/Runner.Worker/OSWarningChecker.cs (hunk 1 above)
Comment (ericsciple) [reply src/Runner.Worker/OSWarningChecker.cs:105]: might be enough after publishing non-trimmed

**Review by TingluoHuang — APPROVED** [review]

**Review by SidhantaKanha — COMMENTED** [review]
#

**Pull request merged**